src/main/scala/encry/local/miner/Miner.scala

```diff
@@ -47,43 +48,49 @@ class Miner extends Actor with Logging {
     !candidateOpt.flatMap(_.parentOpt).map(_.id).exists(_.sameElements(b.header.id))
 
   def shouldStartMine(b: EncryBlock): Boolean =
-    settings.node.mining && b.header.timestamp >= timeProvider.time() && context.children.nonEmpty
+    settings.node.mining && candidateOpt.forall(_.timestamp <= b.header.timestamp)
```

Comment:
Its reasonable to check here whether `header` is new, condition used here is slightly related to that factor.